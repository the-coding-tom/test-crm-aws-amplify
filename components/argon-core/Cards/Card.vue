<template>
  <div
    :class="[
      {'card-lift--hover': hover},
      {'shadow': shadow},
      {[`shadow-${shadowSize}`]: shadowSize},
      {[`bg-gradient-${gradient}`]: gradient},
      {[`bg-${type}`]: type}
    ]"
    class="card">

    <slot name="image"/>
    <div
      v-if="$slots.header"
      :class="headerClasses"
      class="card-header">
      <slot name="header"/>
    </div>
    <div
      v-if="!noBody"
      :class="bodyClasses"
      class="card-body">
      <slot/>
    </div>

    <slot v-if="noBody"/>

    <div
      v-if="$slots.footer"
      :class="footerClasses"
      class="card-footer">
      <slot name="footer"/>
    </div>
  </div>
</template>
<script>
export default {
  name: 'Card',
  props: {
    type: {
      type: String,
      description: 'Card type',
      default: null
    },
    gradient: {
      type: String,
      description: 'Card background gradient type (warning,danger etc)',
      default: null
    },
    hover: {
      type: Boolean,
      description: 'Whether card should move on hover',
      default: null
    },
    shadow: {
      type: Boolean,
      description: 'Whether card has shadow',
      default: null
    },
    shadowSize: {
      type: String,
      description: 'Card shadow size',
      default: null
    },
    noBody: {
      type: Boolean,
      default: false,
      description: 'Whether card should have wrapper body class'
    },
    bodyClasses: {
      type: [String, Object, Array],
      description: 'Card body css classes',
      default: null
    },
    headerClasses: {
      type: [String, Object, Array],
      description: 'Card header css classes',
      default: null
    },
    footerClasses: {
      type: [String, Object, Array],
      description: 'Card footer css classes',
      default: null
    }
  }
}
</script>
<style>
</style>
