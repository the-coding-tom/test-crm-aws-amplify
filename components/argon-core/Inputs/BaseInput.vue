<template>
  <div class="form-group">
    <slot name="label">
      <label
        v-if="label"
        :class="labelClasses">
        {{ label }}
      </label>
    </slot>
    <div
      :class="[
        {'input-group': hasIcon},
        {'focused': focused},
        {'input-group-alternative': alternative},
        {'has-label': label || $slots.label},
        inputGroupClasses
    ]">
      <div
        v-if="prependIcon || $slots.prepend"
        class="input-group-prepend">
        <span class="input-group-text">
          <slot name="prepend">
            <i :class="prependIcon"/>
          </slot>
        </span>
      </div>
      <slot v-bind="slotData">
        <input
          :value="value"
          :type="type"
          v-bind="$attrs"
          :valid="!error"
          :required="required"
          :class="[{'is-valid': valid === true}, {'is-invalid': error}, inputClasses]"
          class="form-control"
          v-on="listeners">
      </slot>
      <div
        v-if="appendIcon || $slots.append"
        class="input-group-append">
        <span class="input-group-text">
          <slot name="append">
            <i :class="appendIcon"/>
          </slot>
        </span>
      </div>
      <slot name="infoBlock"/>
      <slot name="error">
        <div
          v-if="error"
          class="invalid-feedback"
          style="display: block;">
          {{ error }}
        </div>
      </slot>
      <slot name="success">
        <div
          v-if="!error && valid"
          class="valid-feedback">
          {{ successMessage }}
        </div>
      </slot>
    </div>
  </div>
</template>
<script>
export default {
  name: 'BaseInput',
  inheritAttrs: false,
  props: {
    required: {
      type: Boolean,
      description: 'Whether input is required (adds an asterix *)',
      default: true
    },
    group: {
      type: Boolean,
      description:
        'Whether input is an input group (manual override in special cases)',
      default: null
    },
    valid: {
      type: Boolean,
      description: 'Whether is valid',
      default: undefined
    },
    alternative: {
      type: Boolean,
      description: 'Whether input is of alternative layout',
      default: false
    },
    label: {
      type: String,
      description: 'Input label (text before input)',
      default: null
    },
    error: {
      type: String,
      description: 'Input error (below input)',
      default: null
    },
    successMessage: {
      type: String,
      description: 'Input success message',
      default: 'Looks good!'
    },
    labelClasses: {
      type: String,
      description: 'Input label css classes',
      default: 'form-control-label'
    },
    inputClasses: {
      type: String,
      description: 'Input css classes',
      default: null
    },
    inputGroupClasses: {
      type: String,
      description: 'Input group css classes',
      default: null
    },
    value: {
      type: [String, Number],
      description: 'Input value',
      default: null
    },
    type: {
      type: String,
      description: 'Input type',
      default: 'text'
    },
    appendIcon: {
      type: String,
      description: 'Append icon (right)',
      default: null
    },
    prependIcon: {
      type: String,
      description: 'Prepend icon (left)',
      default: null
    }
  },
  data() {
    return {
      focused: false
    }
  },
  computed: {
    listeners() {
      return {
        ...this.$listeners,
        input: this.updateValue,
        focus: this.onFocus,
        blur: this.onBlur
      }
    },
    slotData() {
      return {
        focused: this.focused,
        error: this.error,
        ...this.listeners
      }
    },
    hasIcon() {
      const { append, prepend } = this.$slots
      return (
        append !== undefined ||
        prepend !== undefined ||
        this.appendIcon !== undefined ||
        this.prependIcon !== undefined ||
        this.group
      )
    }
  },
  methods: {
    updateValue(evt) {
      let value = evt.target.value
      this.$emit('input', value)
    },
    onFocus(evt) {
      this.focused = true
      this.$emit('focus', evt)
    },
    onBlur(evt) {
      this.focused = false
      this.$emit('blur', evt)
    }
  }
}
</script>
<style>
</style>
