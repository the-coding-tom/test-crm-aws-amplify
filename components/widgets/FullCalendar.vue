<template>
  <FullCalendar
    :plugins="calendarPlugins"
    :header="{
      left: 'prev,next today',
      center: 'title',
      right: right
    }"
    :views="views"
    :events="events"
    :default-view="defaultView"
    :dates-render="datesRender"
    @dateClick="dateClick"
    @eventClick="eventClick" />
</template>

<script>
import FullCalendar from '@fullcalendar/vue'
import listPlugin from '@fullcalendar/list'
import dayGridPlugin from '@fullcalendar/daygrid'
import timeGridPlugin from '@fullcalendar/timegrid'
import interactionPlugin from '@fullcalendar/interaction'

export default {
  components: {
    FullCalendar // make the <FullCalendar> tag available
  },
  props: {
    events: {
      type: Array,
      default: () => []
    },
    defaultView: {
      type: String,
      default: 'listWeek'
    },
    right: {
      type: String,
      default: 'dayGridMonth,timeGridWeek,timeGridDay,listWeek'
    },
    views: {
      type: Object,
      default: () => ({
        listWeek: {
          noEventsMessage: 'No events to display'
        }
      })
    }
  },
  data() {
    return {
      calendarPlugins: [
        dayGridPlugin,
        timeGridPlugin,
        interactionPlugin,
        listPlugin
      ]
    }
  },
  methods: {
    datesRender({ view }) {
      this.$emit('dateChange', {
        startDate: view.activeStart,
        endDate: view.activeEnd
      })
    },
    eventClick({ event }) {
      this.$emit('eventClick', event)
    },
    dateClick(event) {
      this.$emit('dateClick', event)
    }
  }
}
</script>

<style lang="css">
.fc-event .fc-title {
  color: gray-dark !important;
}
</style>

<style lang="scss">
@import '@fullcalendar/core/main.css';
@import '@fullcalendar/daygrid/main.css';
@import '@fullcalendar/timegrid/main.css';
@import '@fullcalendar/list/main.css';
@import '~/assets/sass/core/vendors/_fullcalendar.scss';
</style>
