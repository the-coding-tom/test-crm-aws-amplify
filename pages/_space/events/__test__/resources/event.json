{
  "id": "8880f445-122a-45bb-9476-a615d7492881",
  "name": "SHACK15 CONVERSATIONS: LISA KRISTINE",
  "description": "A CONVERSATION AND GUIDED ART GALLERY TOUR WITH HUMANITARIAN PHOTOGRAPHER LISA KRISTINE. Join us for a SHACK15 Conversation with Lisa Kristine, an internationally acclaimed humanitarian photographer and social entrepreneur who has documented indigenous peoples and social causes in more than 150 countries across six continents.",
  "banner_url": "https://d3mym26vamg9ol.cloudfront.net/event/1593010166_d2b06fae-7f1c-43fc-b9d3-1f7ec9493beb.png",
  "host_logo": "https://d3mym26vamg9ol.cloudfront.net/event/1593010176_d743ff8d-28c1-4636-8d6d-f2a8140599a2.png",
  "host_name": null,
  "type": "event",
  "start_time": "2020-06-30 14:00:00",
  "end_time": "2020-06-30 15:00:00",
  "price": 10,
  "total_tickets": 10,
  "max_ticket_per_person": 3,
  "send_email": false,
  "email_subject": "Booking Confirmed",
  "email_content": null,
  "attendees_count": 0,
  "event_category": {
    "id": "8e42f835-74b2-475f-b7f3-cf72c6b792ed",
    "name": "talks",
    "color": "#83ac0c",
    "disable": false,
    "created_at": "2019-11-05 08:49:10",
    "updated_at": "2019-11-05 08:49:10"
  },
  "room": null,
  "external_location": "Lounge",
  "related_attendees": [],
  "banner_image": "https://d3mym26vamg9ol.cloudfront.net/event/1593010170_53b0a7de-b4b5-4018-82cb-4468088d4001.png"
}
