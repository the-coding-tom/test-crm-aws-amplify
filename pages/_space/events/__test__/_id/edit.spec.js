import IdIndexPage from '@/pages/_space/events/_id/index.vue'
import IdEditPage from '@/pages/_space/events/_id/edit.vue'
import Card from '~/components/argon-core/Cards/Card.vue'
import BootstrapVue from 'bootstrap-vue'
import BasePagination from '~/components/argon-core/BasePagination'
import BaseInput from '~/components/argon-core/Inputs/BaseInput.vue'
import FullCalendar from '@/components/widgets/FullCalendar'
import { Editor } from '@toast-ui/vue-editor'

import VueRouter from 'vue-router'
import cloneDeep from 'lodash/cloneDeep'
import flushPromises from 'flush-promises'
import { createWrapper } from '@/test.util.js'
import moment from 'moment'

import eventCategories from '../resources/all-categories.json'
import event from '../resources/event.json'
import currentSpace from '@/pages/_space/__test__/resources/current-space.json'

describe('Page: /:space/events/:id/edit | Events Edit Page', () => {
  let NuxtStore, store, wrapper, $event, router

  beforeAll(async () => {
    NuxtStore = await import(`${process.env.buildDir}/store.js`)
    const { createRouter } = await import(`${process.env.buildDir}/router.js`)

    createRouter()

    $event = {
      getEvent: jest.fn().mockResolvedValue({ data: event }),
      getEventCategories: jest
        .fn()
        .mockResolvedValue(cloneDeep(eventCategories)),
      updateEvent: jest.fn().mockResolvedValue({ data: event })
    }

    // mocking 'scrollIntoView' for tui-editor
    window.HTMLElement.prototype.scrollIntoView = jest.fn()
  })

  beforeEach(async () => {
    store = await NuxtStore.createStore()
    store.commit('space/setSpace', currentSpace)

    router = new VueRouter({
      routes: [
        {
          name: 'space-events-id',
          path: '_space/events/:id',
          component: IdIndexPage
        }
      ]
    })

    const params = {
      id: event.id
    }

    const asyncDataResponse = await IdEditPage.asyncData({
      store,
      $event,
      params
    })

    wrapper = createWrapper(
      IdEditPage,
      store,
      router,
      {
        ...asyncDataResponse
      },
      {
        stubs: {
          'nuxt-link': true,
          'client-only': true,
          transition: false,
          'date-picker': true
        }
      },
      vueInstance => {
        vueInstance.use(BootstrapVue)
        vueInstance.component(Card.name, Card)
        vueInstance.component(BasePagination.name, BasePagination)
        vueInstance.component(BaseInput.name, BaseInput)
        vueInstance.component('full-calendar', FullCalendar)
        vueInstance.component('Editor', Editor)
        vueInstance.prototype.$event = $event
        vueInstance.prototype.$moment = moment
      }
    )
  })

  it('renders properly', async () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('shows event info', () => {
    expect(wrapper.find('#title').element.value).toBe(event.name)
    expect(parseInt(wrapper.find('#capacity').element.value)).toBe(
      event.total_tickets
    )
    // expect(wrapper.find('#description').element.value).toBe(
    //   event.description
    // )
    expect(wrapper.vm.$data.event.start_time).toBe(event.start_time)
    expect(wrapper.vm.$data.event.end_time).toBe(event.end_time)
    expect(parseInt(wrapper.find('#price').element.value)).toBe(event.price)
    expect(parseInt(wrapper.find('#maxTicketPerPerson').element.value)).toBe(
      event.max_ticket_per_person
    )
  })

  it('updates event info', async () => {
    const updatedInfo = {
      name: 'Wrap Ahead',
      capacity: 1000,
      price: 1
    }

    wrapper.find('#title').setValue(updatedInfo.name)
    wrapper.find('#capacity').setValue(updatedInfo.capacity)
    wrapper.find('#price').setValue(updatedInfo.price)
    await wrapper.vm.$nextTick()

    wrapper.find('form').trigger('submit.prevent')
    await flushPromises()

    expect($event.updateEvent).toHaveBeenCalled()
    expect($event.updateEvent.mock.calls[0][1].name).toBe(updatedInfo.name)
    expect(parseInt($event.updateEvent.mock.calls[0][1].price)).toBe(
      updatedInfo.price
    )
    expect(parseInt($event.updateEvent.mock.calls[0][1].total_tickets)).toBe(
      updatedInfo.capacity
    )
  })
})
