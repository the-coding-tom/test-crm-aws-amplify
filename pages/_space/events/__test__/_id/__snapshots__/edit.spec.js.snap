// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page: /:space/events/:id/edit | Events Edit Page renders properly 1`] = `
<div>
  <form class="">
    <div class="header pb-6">
      <div class="container-fluid">
        <div class="header-body">
          <div class="d-flex justify-content-between align-items-center py-4">
            <h1 class="page-title"><span>Event</span> <b>â€¢</b> Update</h1>
            <div><button type="submit" class="btn btn btn-primary text-white btn-secondary">Update Event</button> <button type="button" class="btn text-danger btn-transparent"><i class="fa fa-angle-left"></i> Cancel Update</button></div>
          </div>
        </div>
      </div>
    </div>
    <div class="container-fluid mt--6">
      <div class="card-deck flex-column flex-xl-row">
        <div class="card">
          <!---->
          <div class="card-body">
            <h3 class="mr-b-20">Event Details</h3>
            <div class="row">
              <div class="col-md-6">
                <div class="row pd-r-20">
                  <div class="form-group col-md-6"><label class="form-control-label">
                      Event Title
                    </label>
                    <div class="input-group has-label">
                      <!----> <input type="text" valid="true" required="required" id="title" placeholder="Awesome Co" class="form-control">
                      <!---->
                      <!---->
                      <!---->
                    </div>
                  </div>
                  <div class="form-group col-md-6"><label class="form-control-label">
                      Capacity
                    </label>
                    <div class="input-group has-label">
                      <!----> <input type="number" valid="true" required="required" id="capacity" placeholder="0" class="form-control">
                      <!---->
                      <!---->
                      <!---->
                    </div>
                  </div>
                  <div class="form-group col-md-12"><label>Event Description</label>
                    <div id="description" placeholder="Add details about the event">
                      <client-only-stub>
                        <div value="A CONVERSATION AND GUIDED ART GALLERY TOUR WITH HUMANITARIAN PHOTOGRAPHER LISA KRISTINE. Join us for a SHACK15 Conversation with Lisa Kristine, an internationally acclaimed humanitarian photographer and social entrepreneur who has documented indigenous peoples and social causes in more than 150 countries across six continents." style="box-sizing: border-box; height: 300px;">
                          <div class="tui-editor-defaultUI">
                            <div class="te-toolbar-section">
                              <div class="te-markdown-tab-section" style="display: none;">
                                <div class="te-tab"><button type="button" data-index="0" class="te-tab-active">Write</button><button type="button" data-index="1">Preview</button></div>
                              </div>
                              <div class="tui-editor-defaultUI-toolbar"><button class="tui-heading tui-toolbar-icons" type="button"></button><button class="tui-bold tui-toolbar-icons" type="button"></button><button class="tui-strike tui-toolbar-icons" type="button"></button>
                                <div class="tui-toolbar-divider"></div><button class="tui-hrline tui-toolbar-icons" type="button"></button><button class="tui-quote tui-toolbar-icons" type="button"></button>
                                <div class="tui-toolbar-divider"></div><button class="tui-ul tui-toolbar-icons" type="button"></button><button class="tui-ol tui-toolbar-icons" type="button"></button><button class="tui-task tui-toolbar-icons" type="button"></button><button class="tui-indent tui-toolbar-icons" type="button"></button><button class="tui-outdent tui-toolbar-icons" type="button"></button>
                                <div class="tui-toolbar-divider"></div><button class="tui-table tui-toolbar-icons" type="button"></button><button class="tui-image tui-toolbar-icons" type="button"></button><button class="tui-link tui-toolbar-icons" type="button"></button>
                                <div class="tui-toolbar-divider"></div><button class="tui-code tui-toolbar-icons" type="button"></button><button class="tui-codeblock tui-toolbar-icons" type="button"></button>
                                <div class="tui-toolbar-divider" style="display: none;"></div><button class="tui-scrollsync active" type="button" style="display: none;"></button>
                                <div class="tui-popup-wrapper te-dropdown-toolbar" style="display: none;">
                                  <div class="tui-popup-body">
                                    <div class="tui-editor-defaultUI-toolbar"></div>
                                  </div>
                                </div>
                                <div class="tui-popup-wrapper te-popup-add-table" style="display: none; position: absolute; width: 181px;">
                                  <div class="tui-popup-body">
                                    <div class="te-table-selection">
                                      <div class="te-table-header" style="height: 17px; width: 151px;"></div>
                                      <div class="te-table-body" style="height: 120px; width: 151px;"></div>
                                      <div class="te-selection-area"></div>
                                    </div>
                                    <p class="te-description"></p>
                                  </div>
                                </div>
                                <div class="tui-popup-wrapper te-heading-add" style="display: none; position: absolute;">
                                  <div class="tui-popup-body">
                                    <ul>
                                      <li data-value="1" data-type="Heading">
                                        <h1>Heading 1</h1>
                                      </li>
                                      <li data-value="2" data-type="Heading">
                                        <h2>Heading 2</h2>
                                      </li>
                                      <li data-value="3" data-type="Heading">
                                        <h3>Heading 3</h3>
                                      </li>
                                      <li data-value="4" data-type="Heading">
                                        <h4>Heading 4</h4>
                                      </li>
                                      <li data-value="5" data-type="Heading">
                                        <h5>Heading 5</h5>
                                      </li>
                                      <li data-value="6" data-type="Heading">
                                        <h6>Heading 6</h6>
                                      </li>
                                      <li data-type="Paragraph">
                                        <div>Paragraph</div>
                                      </li>
                                    </ul>
                                  </div>
                                </div>
                              </div>
                            </div>
                            <div class="te-editor-section">
                              <div class="tui-editor te-ww-mode">
                                <div class="te-md-container te-preview-style-tab">
                                  <div class="te-editor te-tab-active"><textarea style="display: none;"></textarea>
                                    <div class="CodeMirror cm-s-default CodeMirror-wrap" style="min-height: 300px;">
                                      <div style="overflow: hidden; position: relative; width: 3px; height: 0px;"><textarea style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;" autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0"></textarea></div>
                                      <div class="CodeMirror-vscrollbar" tabindex="-1" cm-not-content="true">
                                        <div style="min-width: 1px;"></div>
                                      </div>
                                      <div class="CodeMirror-hscrollbar" tabindex="-1" cm-not-content="true">
                                        <div style="height: 100%; min-height: 1px;"></div>
                                      </div>
                                      <div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div>
                                      <div class="CodeMirror-gutter-filler" cm-not-content="true"></div>
                                      <div class="CodeMirror-scroll" tabindex="-1">
                                        <div class="CodeMirror-sizer" style="margin-left: 0px;">
                                          <div style="position: relative;">
                                            <div class="CodeMirror-lines" role="presentation">
                                              <div style="position: relative; outline: none;" role="presentation">
                                                <div class="CodeMirror-measure"><pre class="CodeMirror-line-like"><span>xxxxxxxxxx</span></pre>
                                                </div>
                                                <div class="CodeMirror-measure"></div>
                                                <div style="position: relative; z-index: 1;"></div>
                                                <div class="CodeMirror-cursors"></div>
                                                <div class="CodeMirror-code" role="presentation"></div>
                                              </div>
                                            </div>
                                          </div>
                                        </div>
                                        <div style="position: absolute; height: 50px; width: 1px;"></div>
                                        <div class="CodeMirror-gutters" style="display: none;"></div>
                                      </div>
                                    </div>
                                  </div>
                                  <div class="te-md-splitter"></div>
                                  <div class="te-preview" style="min-height: 300px;">
                                    <div class="tui-editor-contents"></div>
                                  </div>
                                </div>
                                <div class="te-ww-container">
                                  <div class="te-editor" style="position: relative;">
                                    <div contenteditable="true" class="tui-editor-contents tui-editor-contents-placeholder" style="min-height: 300px;">
                                      <div><br></div>
                                    </div>
                                    <div class="te-ww-block-overlay code-block-header" style="position: absolute; display: none; z-index: 1;"><span>text</span><button type="button">Editor</button></div>
                                  </div>
                                </div>
                              </div>
                            </div>
                            <div class="te-mode-switch-section" style="display: none;">
                              <div class="te-mode-switch"><button class="te-switch-button markdown" type="button">Markdown</button><button class="te-switch-button wysiwyg active" type="button">WYSIWYG</button></div>
                            </div>
                            <div class="tui-popup-wrapper te-popup-add-link tui-editor-popup" style="display: none;">
                              <div class="tui-popup-header">
                                <span class="tui-popup-title">Insert link</span>
                                <div class="tui-popup-header-buttons">
                                  <button type="button" class="tui-popup-close-button"></button>
                                </div>
                              </div>
                              <div class="tui-popup-body">
                                <label for="url">URL</label>
                                <input type="text" class="te-url-input">
                                <label for="linkText">Link text</label>
                                <input type="text" class="te-link-text-input">
                                <div class="te-button-section">
                                  <button type="button" class="te-ok-button">OK</button>
                                  <button type="button" class="te-close-button">Cancel</button>
                                </div>
                              </div>
                            </div>
                            <div class="tui-popup-wrapper te-popup-add-image tui-editor-popup" style="display: none;">
                              <div class="tui-popup-header">
                                <span class="tui-popup-title">Insert image</span>
                                <div class="tui-popup-header-buttons">
                                  <button type="button" class="tui-popup-close-button"></button>
                                </div>
                              </div>
                              <div class="tui-popup-body">
                                <div class="te-tab-section">
                                  <div class="te-tab"><button type="button" data-index="0" class="te-tab-active">File</button><button type="button" data-index="1">URL</button></div>
                                </div>
                                <div class="te-url-type">
                                  <label for="">Image URL</label>
                                  <input type="text" class="te-image-url-input">
                                </div>
                                <div class="te-file-type te-tab-active">
                                  <label for="">Select image file</label>
                                  <input type="file" class="te-image-file-input" accept="image/*">
                                </div>
                                <label for="url">Description</label>
                                <input type="text" class="te-alt-text-input">
                                <div class="te-button-section">
                                  <button type="button" class="te-ok-button">OK</button>
                                  <button type="button" class="te-close-button">Cancel</button>
                                </div>
                              </div>
                            </div>
                            <div class="tui-popup-wrapper te-popup-table-utils" style="display: none;">
                              <div class="tui-popup-body">
                                <button type="button" class="te-table-add-row">Add row</button>
                                <button type="button" class="te-table-add-col">Add col</button>
                                <button type="button" class="te-table-remove-row">Remove row</button>
                                <button type="button" class="te-table-remove-col">Remove col</button>
                                <hr>
                                <button type="button" class="te-table-col-align-left">Align left</button>
                                <button type="button" class="te-table-col-align-center">Align center</button>
                                <button type="button" class="te-table-col-align-right">Align right</button>
                                <hr>
                                <button type="button" class="te-table-remove">Remove table</button>
                              </div>
                            </div>
                            <div class="tui-popup-wrapper te-popup-code-block-languages" style="display: none; z-index: 10000;">
                              <div class="tui-popup-body"></div>
                            </div>
                            <div class="tui-popup-modal-background tui-popup-code-block-editor" style="display: none;">
                              <div class="tui-popup-wrapper">
                                <div class="tui-popup-header">
                                  <span class="tui-popup-title">CodeBlock Editor<span class="te-input-language" style="display: none;"><input type="text" maxlength="20" placeholder="Choose language"></span></span>
                                  <div class="tui-popup-header-buttons">
                                    <button type="button" class="popup-editor-toggle-scroll active" style="display: inline-block;"></button>
                                    <button type="button" class="popup-editor-toggle-preview active"></button>
                                    <button type="button" class="popup-editor-toggle-fit"></button>
                                    <button type="button" class="tui-popup-close-button"></button>
                                  </div>
                                </div>
                                <div class="tui-popup-body">
                                  <div class="popup-editor-body">
                                    <div class="tui-split-scroll scroll-sync">
                                      <div class="tui-split-scroll-wrapper">
                                        <div class="tui-split-scroll-content">
                                          <div class="popup-editor-editor-wrapper tui-split-content-left"><textarea style="display: none;"></textarea>
                                            <div class="CodeMirror cm-s-default CodeMirror-wrap">
                                              <div style="overflow: hidden; position: relative; width: 3px; height: 0px;"><textarea style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;" autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0"></textarea></div>
                                              <div class="CodeMirror-vscrollbar" tabindex="-1" cm-not-content="true">
                                                <div style="min-width: 1px;"></div>
                                              </div>
                                              <div class="CodeMirror-hscrollbar" tabindex="-1" cm-not-content="true">
                                                <div style="height: 100%; min-height: 1px;"></div>
                                              </div>
                                              <div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div>
                                              <div class="CodeMirror-gutter-filler" cm-not-content="true"></div>
                                              <div class="CodeMirror-scroll" tabindex="-1">
                                                <div class="CodeMirror-sizer" style="margin-left: 0px;">
                                                  <div style="position: relative;">
                                                    <div class="CodeMirror-lines" role="presentation">
                                                      <div style="position: relative; outline: none;" role="presentation">
                                                        <div class="CodeMirror-measure"><pre class="CodeMirror-line-like"><span>xxxxxxxxxx</span></pre>
                                                        </div>
                                                        <div class="CodeMirror-measure"></div>
                                                        <div style="position: relative; z-index: 1;"></div>
                                                        <div class="CodeMirror-cursors"></div>
                                                        <div class="CodeMirror-code" role="presentation"></div>
                                                      </div>
                                                    </div>
                                                  </div>
                                                </div>
                                                <div style="position: absolute; height: 50px; width: 1px;"></div>
                                                <div class="CodeMirror-gutters" style="display: none;"></div>
                                              </div>
                                            </div>
                                          </div>
                                          <div class="tui-split-content-right" style="top: 0px;">
                                            <div class="tui-editor-contents"></div>
                                          </div>
                                        </div>
                                        <div class="tui-splitter"></div>
                                      </div>
                                    </div>
                                  </div>
                                  <div class="te-button-section">
                                    <button type="button" class="te-ok-button">OK</button>
                                    <button type="button" class="te-close-button">Cancel</button>
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </client-only-stub>
                    </div>
                  </div>
                  <fieldset class="form-group col-md-6" id="__BVID__14">
                    <legend tabindex="-1" class="bv-no-focus-ring col-form-label pt-0" id="__BVID__14__BV_label_">Start Date</legend>
                    <div tabindex="-1" role="group" class="bv-no-focus-ring">
                      <client-only-stub>
                        <date-picker-stub id="startTime" width="100%" input-class="form-control" lang="en" format="YYYY-MM-DD HH:mm:ss" value-type="format" confirm="" type="datetime" placeholder="Start Date" value="2020-06-30 14:00:00"></date-picker-stub>
                      </client-only-stub>
                      <!---->
                      <!---->
                      <!---->
                    </div>
                  </fieldset>
                  <fieldset class="form-group col-md-6" id="__BVID__17">
                    <legend tabindex="-1" class="bv-no-focus-ring col-form-label pt-0" id="__BVID__17__BV_label_">End Date</legend>
                    <div tabindex="-1" role="group" class="bv-no-focus-ring">
                      <client-only-stub>
                        <date-picker-stub id="endTime" width="100%" input-class="form-control" lang="en" format="YYYY-MM-DD HH:mm:ss" value-type="format" confirm="" type="datetime" placeholder="End Date" value="2020-06-30 15:00:00"></date-picker-stub>
                      </client-only-stub>
                      <!---->
                      <!---->
                      <!---->
                    </div>
                  </fieldset>
                  <div class="form-group col-md-6"><label class="form-control-label">
                      Price
                    </label>
                    <div class="input-group has-label">
                      <!----> <input type="number" valid="true" required="required" id="price" step="0.01" placeholder="0.00" class="form-control">
                      <!---->
                      <!---->
                      <!---->
                    </div>
                  </div>
                  <div class="form-group col-md-6"><label class="form-control-label">
                      Max Ticket Per Person
                    </label>
                    <div class="input-group has-label">
                      <!----> <input type="number" valid="true" required="required" id="maxTicketPerPerson" placeholder="3" class="form-control">
                      <!---->
                      <!---->
                      <!---->
                    </div>
                  </div>
                </div>
              </div>
              <div class="col-md-6">
                <div class="row pd-l-20">
                  <div class="form-group col-md-12"><label>Event Category:</label>
                    <div class="el-select">
                      <!---->
                      <div class="el-input el-input--suffix">
                        <!----><input type="text" readonly="readonly" autocomplete="off" placeholder="Select Category" class="el-input__inner">
                        <!----><span class="el-input__suffix"><span class="el-input__suffix-inner"><i class="el-select__caret el-input__icon el-icon-arrow-up"></i><!----><!----><!----><!----><!----></span>
                        <!----></span>
                        <!---->
                        <!---->
                      </div>
                      <div class="el-select-dropdown el-popper" style="display: none;">
                        <div class="el-scrollbar" style="">
                          <div class="el-select-dropdown__wrap el-scrollbar__wrap el-scrollbar__wrap--hidden-default">
                            <ul class="el-scrollbar__view el-select-dropdown__list">
                              <!---->
                              <li class="el-select-dropdown__item selected"><span>talks</span></li>
                              <li class="el-select-dropdown__item"><span>meetups</span></li>
                            </ul>
                          </div>
                          <div class="el-scrollbar__bar is-horizontal">
                            <div class="el-scrollbar__thumb" style="transform: translateX(0%); webkit-transform: translateX(0%);"></div>
                          </div>
                          <div class="el-scrollbar__bar is-vertical">
                            <div class="el-scrollbar__thumb" style="transform: translateY(0%); webkit-transform: translateY(0%);"></div>
                          </div>
                        </div>
                        <!---->
                      </div>
                    </div>
                  </div>
                  <div class="form-group col-md-12">
                    <div class="custom-control custom-checkbox"><input id="external-1" type="checkbox" name="external-1" autocomplete="off" class="custom-control-input" value="true"><label for="external-1" class="custom-control-label">
                        Host at an external location
                      </label></div> <input type="text" placeholder="External Location" required="required" aria-required="true" class="form-control" id="__BVID__32">
                  </div>
                  <!---->
                  <div class="form-group col-md-12">
                    <div class="custom-control custom-checkbox"><input id="sendEmailCheckbox" type="checkbox" autocomplete="off" class="custom-control-input" value="true"><label for="sendEmailCheckbox" class="custom-control-label">Send email to attendees</label></div>
                  </div>
                  <!---->
                </div>
              </div>
            </div>
            <div>
              <div id="preview" class="mr-b-20"><img src="https://d3mym26vamg9ol.cloudfront.net/event/1593010166_d2b06fae-7f1c-43fc-b9d3-1f7ec9493beb.png"></div>
              <div class="bg-lgray pd-10 pd-l-20 upload-btn-wrapper pd-0"><button class="btn"><i class="fa fa-paperclip mr-r-10"></i> Upload Event Image (&lt;500KB &amp; size 1125x582)</button> <input name="eventbanner" type="file" accept="image/*"></div>
            </div>
            <div value="https://d3mym26vamg9ol.cloudfront.net/event/1593010170_53b0a7de-b4b5-4018-82cb-4468088d4001.png">
              <div id="preview" class="mr-b-20"><img src="https://d3mym26vamg9ol.cloudfront.net/event/1593010170_53b0a7de-b4b5-4018-82cb-4468088d4001.png"></div>
              <div class="bg-lgray pd-10 pd-l-20 upload-btn-wrapper pd-0"><button class="btn"><i class="fa fa-paperclip mr-r-10"></i> Upload Event Banner (&lt;500KB &amp; size 1125x582)</button> <input name="eventbanner" type="file" accept="image/*"></div>
            </div>
            <div value="https://d3mym26vamg9ol.cloudfront.net/event/1593010176_d743ff8d-28c1-4636-8d6d-f2a8140599a2.png">
              <div id="preview" class="mr-b-20"><img src="https://d3mym26vamg9ol.cloudfront.net/event/1593010176_d743ff8d-28c1-4636-8d6d-f2a8140599a2.png"></div>
              <div class="bg-lgray pd-10 pd-l-20 upload-btn-wrapper pd-0"><button class="btn"><i class="fa fa-paperclip mr-r-10"></i> Upload Host Logo (&lt;500KB &amp; size 500x500)</button> <input name="hostlogo" type="file" accept="image/*"></div>
            </div>
          </div>
          <!---->
          <!---->
        </div>
      </div>
    </div>
  </form>
</div>
`;
